src/UI/StaffCraftingMenu.cpp

```diff
@@ -328,14 +348,37 @@ namespace UI
 	{
 		if (selected.morpholith) {
 			const auto morpholithCharge = GetEntryDataSoulCharge(selected.morpholith->data.get());
+			const auto morpholithCount = selected.morpholith->data->countDelta;
 			return morpholithCharge > 0.0f
 				? morpholithCharge >= CalculateSpellCost(a_spell)
-				: heartStoneCount >= GetSpellHeartstones(a_spell);
+				: morpholithCount >= GetSpellHeartstones(a_spell);
 		}
-		return heartStoneCount >= GetSpellHeartstones(a_spell) ||
+		return highestMorpholithCount >= GetSpellHeartstones(a_spell) ||
 			maxSoulSize >= CalculateSpellCost(a_spell);
 	}
 
+	bool StaffCraftingMenu::IsValidMorpholith(
+		const RE::BGSKeywordForm* a_obj,
+		const std::vector<RE::BGSKeyword*>& a_vec)
+	{
+		if (a_vec.empty()) {
+			return false;
+		}
+
+		for (const RE::BGSKeyword* const keyword :
+			 std::span(a_obj->keywords, a_obj->numKeywords)) {
+			if (!keyword)
+				continue;
+
+			for (const auto recordedKeyword : a_vec) {
+				if (recordedKeyword == keyword) {
+					return true;
+				}
+			}
+		}
```

Comment:
If we're just doing a pointer comparison anyway, I would prefer to call `HasKeyword` here.